File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        left = 0
        result = 0
        visited=[False]*256
        for right in range(len(s)):
            while visited[ord(s[right])]:
                visited[ord(s[left])]=False
                left+=1
            visited[ord(s[right])]=True
            result=max(result,right-left+1)
        return result

        

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int result = 0;
        Set<Character>set=new HashSet<>();
        int l=0;
        for(int r=0; r<s.length(); r++){
            char c = s.charAt(r);
            while(set.contains(c)){
                set.remove(s.charAt(l));
                l++;
            }
            set.add(c);
            result = Math.max(result,set.size());
        }
        return result;

        
    }
}